backend/lightning/lightning.go

```diff
@@ -155,34 +165,60 @@ func (lightning *Lightning) connect() {
 		// support multiple accounts, for future extensions.
 		account := lightningConfig.Accounts[0]
 
-		// TODO: this seed should be determined from the account/device.
 		seed, err := breez_sdk.MnemonicToSeed(account.Mnemonic)
-
 		if err != nil {
-			lightning.log.WithError(err).Warn("BreezSDK: MnemonicToSeed failed")
+			lightning.log.WithError(err).Error("BreezSDK: MnemonicToSeed failed")
 			return
 		}
 
+		var greenlightCredentials *breez_sdk.GreenlightCredentials
+		if registerNode {
+			statusCode, deviceKey, err := util.HTTPGet(lightning.httpClient, greenLightKeyUrl, "", int64(4096))
+			if err != nil {
+				lightning.log.WithError(err).Errorf("Greenlight key fetch failed, errorCode: %v", statusCode)
+				return
+			}
+
+			statusCode, deviceCert, err := util.HTTPGet(lightning.httpClient, greenLightCertUrl, "", int64(4096))
+			if err != nil {
+				lightning.log.WithError(err).Errorf("Greenlight cert fetch failed, errorCode: %v", statusCode)
+				return
+			}
+
+			greenlightCredentials = &breez_sdk.GreenlightCredentials{
+				DeviceKey:  deviceKey,
+				DeviceCert: deviceCert,
+			}
+		}
+
 		nodeConfig := breez_sdk.NodeConfigGreenlight{
 			Config: breez_sdk.GreenlightNodeConfig{
-				PartnerCredentials: nil,
+				PartnerCredentials: greenlightCredentials,
 				InviteCode:         nil,
 			},
 		}
 
 		workingDir := path.Join(lightning.cacheDirectoryPath, accountBreezFolder(account.Code))
 
 		if err := os.MkdirAll(workingDir, 0700); err != nil {
-			lightning.log.WithError(err).Warn("Error creating working directory")
+			lightning.log.WithError(err).Error("Error creating working directory")
 			return
 		}
 
-		config := breez_sdk.DefaultConfig(breez_sdk.EnvironmentTypeProduction, "", nodeConfig)
+		statusCode, breezApiKey, err := util.HTTPGet(lightning.httpClient, breezApiKeyUrl, "", int64(4096))
+		if err != nil {
+			lightning.log.WithError(err).Errorf("Breez api key fetch failed, errorCode: %v", statusCode)
```

Comment:
i think the status code is already in the error message of `err`, isn't it? no need to repeat it. same above